src/services/terms.ts

```diff
@@ -1,20 +1,38 @@
 import * as RDF from 'rdf-js';
 
 export class Term {
-  id: RDF.Term | undefined = undefined;
+  constructor(
+    readonly id: RDF.Term,
+    readonly type: RDF.Term | undefined,
+    readonly prefLabels: RDF.Term[],
+    readonly altLabels: RDF.Term[],
+    readonly hiddenLabels: RDF.Term[],
+    readonly scopeNotes: RDF.Term[],
+    readonly broader: RelatedTerm[],
+    readonly narrower: RelatedTerm[]
+  ) {}
+}
+
+export class RelatedTerm {
+  constructor(readonly id: RDF.Term) {}
+  prefLabels: RDF.Term[] = [];
+}
+
+class SparqlResultTerm {
+  constructor(readonly id: RDF.Term) {}
   type: RDF.Term | undefined = undefined;
   prefLabels: RDF.Term[] = [];
   altLabels: RDF.Term[] = [];
   hiddenLabels: RDF.Term[] = [];
   scopeNotes: RDF.Term[] = [];
-  broaderLabels: RDF.Term[] = [];
-  narrowerLabels: RDF.Term[] = [];
+  broader: RDF.Term[] = [];
+  narrower: RDF.Term[] = [];
 }
 
 export class TermsTransformer {
-  protected terms: Term[] = [];
-  protected currentTerm: Term = new Term();
-  protected readonly predicateToPropertyMap = new Map<string, string>([
+  private termsIris: string[] = [];
+  private termsMap: Map<string, SparqlResultTerm> = new Map();
```

Comment:
I had to replace the simple `currentTerm` state management with:

- a list that is ordered and tracks all search results (i.e. top-level terms in the SPARQL result)
- a map for quick lookups of related terms.

Reply:
👍🏻